src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ReuploadFailedEnvelopeTask.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import com.netflix.servo.util.ThreadFactories;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Lookup;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscanprocessor.config.EnvelopeAccessProperties;
+import uk.gov.hmcts.reform.bulkscanprocessor.config.EnvelopeAccessProperties.Mapping;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.FailedDocUploadProcessor;
+
+import java.util.List;
+import java.util.concurrent.CompletionService;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+/**
+ * This class is a task executed by Scheduler as per configured interval.
+ * It will retrieve failed to upload envelopes from DB and try to re-upload.
+ * <ol>
+ *     <li>Get batch of failed to upload envelopes</li>
+ *     <li>Retrieve relevant zip file from Blob Storage</li>
+ *     <li>Try to upload pdfs</li>
+ * </ol>
+ */
+@Component
+@ConditionalOnProperty(value = "scheduling.enabled", matchIfMissing = true)
+@EnableConfigurationProperties(EnvelopeAccessProperties.class)
+public class ReuploadFailedEnvelopeTask {
```

Comment:
Are we missing unit tests for this class ?

Reply:
I'll think of something here. Since actual task deals with concurrency and processor - envelopes. Latter is/will be extensively tested in next PR